client/securedrop_client/gui/widgets.py

```diff
@@ -424,6 +427,113 @@ def clear_message(self) -> None:
         self._hide()
 
 
+class InnerTopPane(QWidget):
+    """
+    Top pane of the MainView window. This pane holds the Batch Action layout,
+    and eventually will hold the keyword search/filter by codename bar.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.setObjectName("InnerTopPane")
+
+        # Use a vertical layout so that the keyword search bar can be added later
+        layout = QVBoxLayout(self)
+        layout.setContentsMargins(0, 0, 0, 0)
+        layout.setSpacing(0)
+        layout.setAlignment(Qt.AlignVCenter)
+        self.setLayout(layout)
+        self.setAttribute(Qt.WA_StyledBackground, True)
+
+        self.batch_actions = BatchActionWidget()
+        layout.addWidget(self.batch_actions)
+
+    def setup(self, controller: Controller) -> None:
+        self.batch_actions.setup(controller)
+
+
+class BatchActionWidget(QWidget):
+    def __init__(self) -> None:
+        super().__init__()
+
+        # CSS style id
+        self.setObjectName("BatchActionWidget")
+
+        # Solid background colour
+        self.setAttribute(Qt.WA_StyledBackground, True)
+        layout = QHBoxLayout()
+        self.setLayout(layout)
+
+        self.toolbar = BatchActionToolbar()
+
+        layout.addWidget(self.toolbar)
+        layout.addStretch()
+
+    def setup(self, controller: Controller) -> None:
+        self.toolbar.setup(controller)
+
+
+class BatchActionToolbar(QToolBar):
+    """
+    A toolbar that contains batch actions (actions that target multiple
+    sources in the ConversationView, and therefore don't belong in the
+    individual conversation menu). Currently, this widget will hold the
+    "Delete Sources" (batch-delete) action.
+
+    For user-facing naming consistency, these items won't be called
+    "batch/bulk <delete>", but simply "<verb> <noun>s" (eg "Delete Sources"), where
+    the original nomenclature comes from the individual Source overflow QAction menu
+    items. Each item may have a tooltip, visible on hover, that provides a more
+    lengthy explanation (e.g., "Delete multiple source accounts").
```

Comment:
I like the local style guide for "here's how to use this section of the UI".  I'll note for the record that I continue to think that "batch" is an implementation detail of what is really just a "bulk" operation or an operation over multiple objects—not all of which will batch the individual operations or requests.  I'm fine with whatever terminology we adopt as long as we don't wind up confused by it.